cmd/aws-actuator/main.go

```diff
@@ -435,221 +302,92 @@ func bootstrapCommand() *cobra.Command {
+				glog.Errorf("Unable to create aws client: %v", err)
+				return err
 			}
 
-			err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
-				glog.Info("Waiting for all nodes to come up")
-				nodesList, err := kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
-				if err != nil {
-					return false, nil
-				}
-
-				nodesReady := true
-				for _, node := range nodesList.Items {
-					ready := false
-					for _, c := range node.Status.Conditions {
-						if c.Type != apiv1.NodeReady {
-							continue
-						}
-						ready = true
-					}
-					glog.Infof("Is node %q ready?: %v\n", node.Name, ready)
-					if !ready {
-						nodesReady = false
-					}
-				}
-
-				return nodesReady, nil
-			})
+			acw := awsclientwrapper.NewAwsClientWrapper(awsClient)
+			glog.Infof("Collecting master kubeconfig")
+			restConfig, err := f.GetMasterMachineRestConfig(masterMachine, acw)
 			if err != nil {
-				glog.Errorf("Failed to get all nodes up and running: %v", err)
-			}
-
-			glog.Info("Deploying cluster-api stack")
-			glog.Info("Deploying aws credentials")
-
-			if err := createNamespace(tc, "test"); err != nil {
+				glog.Errorf("Unable to pull kubeconfig: %v", err)
 				return err
 			}
 
-			awsCredentials := &apiv1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "aws-credentials-secret",
-					Namespace: "test",
-				},
-				Data: map[string][]byte{
-					"awsAccessKeyId":     []byte(os.Getenv("AWS_ACCESS_KEY_ID")),
-					"awsSecretAccessKey": []byte(os.Getenv("AWS_SECRET_ACCESS_KEY")),
+			clusterFramework, err := framework.NewFrameworkFromConfig(
+				restConfig,
+				&framework.SSHConfig{
+					Key:  mastermachinepk,
+					User: "ec2-user",
 				},
-			}
-
-			if err := createSecret(tc, awsCredentials); err != nil {
-				return err
-			}
-
-			err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
-				glog.Info("Deploying cluster-api server")
-				if output, err := cmdRun("kubectl", "--kubeconfig=kubeconfig", "apply", fmt.Sprintf("-f=%v", path.Join(manifestsDir, "cluster-api-server.yaml")), "--validate=false"); err != nil {
-					glog.Infof("Unable to apply %v manifest: %v\n%v", path.Join(manifestsDir, "cluster-api-server.yaml"), err, string(output))
-					return false, nil
-				}
-
-				return true, nil
-			})
+			)
 			if err != nil {
-				glog.Errorf("Unable to deploy cluster-api server %v", err)
+				return err
 			}
 
-			err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
-				glog.Info("Deploying cluster-api controllers")
-				if output, err := cmdRun("kubectl", "--kubeconfig=kubeconfig", "apply", fmt.Sprintf("-f=%v", path.Join(manifestsDir, "provider-components.yml"))); err != nil {
-					glog.Infof("Unable to apply %v manifest: %v\n%v", path.Join(manifestsDir, "provider-components.yml"), err, string(output))
-					return false, nil
+			clusterFramework.ErrNotExpected = func(err error) {
+				if err != nil {
+					panic(err)
```

Comment:
Why `panic`?

Reply:
Since the `ErrNotExpected` is called from within the framework and once there is an error, the execution must stop. Optionally, we can catch the panic but it looked obsolete. If the bootstrap panics I would expect a stack trace so I can see what and where happened.

Reply:
changed to `glog.Fatal`